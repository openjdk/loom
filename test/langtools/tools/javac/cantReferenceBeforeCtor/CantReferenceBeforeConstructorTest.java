/*
 * @test /nodynamiccopyright/
 * @bug 8270835
 * @summary regression after JDK-8261006
 * @compile/fail/ref=CantReferenceBeforeConstructorTest.out -XDrawDiagnostics CantReferenceBeforeConstructorTest.java
 */

public class CantReferenceBeforeConstructorTest {
    // test case I
    class A extends B {
        A(int i) {}

        class C extends B {
            class D extends S {
                D(float f) {
                    C.super.ref.super(); // ok
                }
            }
        }
    }

    class B {
        B ref;
        class S {}
    }

    // test case II
    class AA extends BB.CC {
        AA() {
            this.super();    // error
        }
    }

    class BB {
        class CC extends BB {
            void m() {
                BB.this.f();
            }
        }
        void f() {}
    }
}
