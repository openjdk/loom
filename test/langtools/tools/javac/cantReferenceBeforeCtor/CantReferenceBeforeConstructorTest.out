CantReferenceBeforeConstructorTest.java:30:13: compiler.err.cant.ref.before.ctor.called: this
1 error
